core/design/src/main/kotlin/edu/stanford/spezi/core/design/views/views/viewModifier/viewState/OperationStateAlert.kt

```diff
@@ -0,0 +1,16 @@
+package edu.stanford.spezi.core.design.views.views.viewModifier.viewState
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import edu.stanford.spezi.core.design.views.views.model.OperationState
+
+@Composable
+fun <State : OperationState> OperationStateAlert(
```

Comment:
We do set the MutableState<ViewState> to ViewState.Idle in the onDismissRequest and in the onClick of the cancel button. If you want, we can change it (or create a separate wrapper) for the usage with a value and an onClose lambda that is called on dismiss, would that work?

Reply:
General idea is to avoid composables owning state and manipulating external states, similar proposal as here: https://github.com/StanfordSpezi/SpeziKt/pull/140#discussion_r1847125977. yes, explicit value and lambdas should be the way to go. it is exactly how compose also provides its components